job-dsl-core/src/main/groovy/javaposse/jobdsl/dsl/helpers/step/XShellContext.groovy

```diff
@@ -0,0 +1,44 @@
+package javaposse.jobdsl.dsl.helpers.step
+
+import javaposse.jobdsl.dsl.AbstractContext
+import javaposse.jobdsl.dsl.JobManagement
+
+class XShellContext extends AbstractContext {
+    String commandLine
+    boolean executeFromWorkingDir = false
+    String regexToKill = null
+    int timeAllocated = 0
+
+    protected XShellContext(JobManagement jobManagement) {
+        super(jobManagement)
+    }
+
+    /**
+     * Adds xshell command.
```

Comment:
Improve the GroovyDoc for all methods. Have a look at the xshell plugin's help texts to get an idea.